cirq/study/result.py

```diff
@@ -150,7 +150,7 @@ def measurements(self) -> Dict[str, np.ndarray]:
 
     @property
     def repetitions(self) -> int:
-        return self.data.shape[0]
+        return len(next(iter(self.measurements.values())))
```

Comment:
Optional: I'd add a comment explaining why we do it this way, just to be sure that no one "cleans it up" later.

Reply:
nit: I'd suggest avoiding `next` like this, because it the iterator happens to be empty it raises `StopIteration` which is a special `BaseException` that can have unintended consequences if you let it propagate up the stack (imagine `repetitions` gets called inside a list comprehension somewhere up the stack). I'll grant that it'd be uncommon to have an empty `Result` object, but I've had to deal with stray `StopIteration`s enough that I'm a bit of a stickler about this :-) Internally we have defined an `iterators.first` function to do what is intended here but raise `ValueError` on empty iterable instead; we should consider doing that in cirq to remove the temptation to reach for `next`.

Reply:
Ran into this almost immediately, despite most simulators refusing to produce an empty result. https://github.com/quantumlib/Cirq/issues/3852